app/components/courses/list.js

```diff
@@ -39,21 +38,22 @@ export default class IliosCourseListComponent extends Component {
     return this.intl.t(translation);
   }
 
-  @task
-  *unlockCourse(course) {
-    const permission = yield this.permissionChecker.canUnlockCourse(course);
+  @action
+  async unlockCourse(course) {
```

Comment:
This should be a task, otherwise if the component gets destroyed while one of the `await`s is going it will cause a complete failure of the app since executing the next line will be on something that doesn't exist anymore.

Reply:
:heavy_check_mark: